code/modules/projectiles/projectile.dm

```diff
@@ -223,7 +223,9 @@
 /obj/item/projectile/Bump(atom/A)
 	var/datum/point/pcache = trajectory.copy_to()
 	var/turf/T = get_turf(A)
-	if(check_ricochet(A) && check_ricochet_flag(A) && ricochets < ricochets_max)
+	if(ricochets >= ricochets_max)

```

Comment:
Sorry, I made a mistake reading this earlier, rather: why was this separated out into a separate if check?

Reply:
first check is required for both ricochet and normal projectiles, then it does an OR to see if it respects the two flags or its a ricochet proj

Reply:
No, why is it in a seperate if statement? This was part of the if statement below originally and you split it out